hydromt/io/writers.py

```diff
@@ -34,41 +30,6 @@ def write_xy(fn, gdf, fmt="%.4f"):
         np.savetxt(f, xy, fmt=fmt)
 
 
-def write_ini_config(
-    config_fn: Union[Path, str],
-    cfdict: dict,
-    encoding: str = "utf-8",
-    cf: ConfigParser = None,
-    noheader: bool = False,
-) -> None:
-    """Write configuration dictionary to ini file.
-
-    Parameters
-    ----------
-    config_fn : Union[Path, str]
-        Path to configuration file
-    cfdict : dict
-        Configuration dictionary.
-    encoding : str, optional
-        File encoding, by default "utf-8"
-    cf : ConfigParser, optional
-        Alternative configuration parser, by default None
-    noheader : bool, optional
-        Set true for a single-level configuration dictionary with no headers,
-        by default False
-    """
-    if cf is None:
-        cf = ConfigParser(allow_no_value=True, inline_comment_prefixes=[";", "#"])
-    elif isinstance(cf, abc.ABCMeta):  # not yet instantiated
-        cf = cf()
-    cf.optionxform = str  # preserve capital letter
-    if noheader:  # add dummy header
-        cfdict = {"dummy": cfdict}
-    cf.read_dict(cfdict)
-    with codecs.open(config_fn, "w", encoding=encoding) as fp:
-        cf.write(fp)
-
-
 def configwrite(config_fn: Union[str, Path], cfdict: dict, **kwargs) -> None:
```

Comment:
This method is used to write model simulation configuration files, e.g. wflow.toml, rather than HydroMT config file. 
For HydroMT build/update config files we want to have only yaml, but we can't decide what file format a model kernel requires to configure a simulation. Since this format is toml in case of most recent model kernels, I suggest to keep the write to toml option.

Reply:
Okay, it is currently really unclear for me which file types should be supported where. Could you please make a table for me with the files we should be able to read and write and which file types should be supported? (excluding data files, that is the adapter part) because I currently can't tell up from down.

Reply:
These are the files:
- HydroMT config: **yaml**
- Model simulation config files:  **toml** by default, but plugins my implement their own format. 
- HydroMT data catalog **yaml**, but this file has its own read and write methods in `DataCatalog` 

So far, we have used the `configread` and `configwrite` functions for both type of config files as these used to be in same format (_ini_) and require the same processing of relative to absolute paths and vice versa. Perhaps its best to move the read and write methods for model simulation configuration **toml** files to the future `ConfigModelComponent` class. 

In that case  `configread` and `configwrite` only need to support HydroMT config **yaml** files.
So basically we need to make a new issue for a `ConfigModelComponent` class to support **toml** and this  PR is fine. Do you agree?

Reply:
yes I think that would be good. I'll make the issue. Could you approve then?

Reply:
see #799